ablog/__init__.py

```diff
@@ -59,6 +72,52 @@ def html_page_context(app, pagename, templatename, context, doctree):
             context["feed_title"] = blog.blog_title
 
 
+def builder_inited(app):
+    if not app.config.inject_templates_after_theme:
+        if not isinstance(app.builder.templates, BuiltinTemplateLoader):
+            raise Exception(
+                "Ablog does not know how to inject templates into with custom "
+                "template bridges. You can use `ablog.get_html_templates_path()` to "
+                "get the path to add in your custom template bridge and set "
+                "`inject_templates_after_theme = False` in your "
+                "`conf.py` file."
+            )
+        if get_html_templates_path() in app.config.templates_path:
+            raise Exception(
+                "Found the path from `ablog.get_html_templates_path()` in the "
+                "`templates_path` variable from `conf.py`. Doing so interferes "
+                "with Ablog's ability to stay compatible with Sphinx themes that "
+                "support it out of the box. Please remove `get_html_templates_path` "
+                "from `templates_path` in your `conf.py` to resolve this."
+            )
```

Comment:
Hmm... this option seems to be doing the wrong thing.

I'd suggest changing this to `skip_injecting_base_ablog_templates` and changing the conf.py documentation related to this as well (I'll make a separate suggestion for that).

The goal isn't to skip doing any checks when that is set, but to avoid doing *anything* when that is set. This would basically be for users who are using a template bridge that does not work like Sphinx's own template bridge does. That's why we have the `isinstance(..., BuiltinTemplateLoader)` check -- since this logic is highly coupled with the way Sphinx's `BuiltinTemplateLoader` works and we can't know if we'd present esoteric errors with alternatives.

```suggestion
    if app.config.skip_injecting_base_ablog_templates:
        return

    if not isinstance(app.builder.templates, BuiltinTemplateLoader):
        raise Exception(
            "Ablog does not know how to inject templates into with custom "
            "template bridges. You can use `ablog.get_html_templates_path()` to "
            "get the path to add in your custom template bridge and set "
            "`skip_injecting_base_ablog_templates = False` in your "
            "`conf.py` file."
        )
    if get_html_templates_path() in app.config.templates_path:
        raise Exception(
            "Found the path from `ablog.get_html_templates_path()` in the "
            "`templates_path` variable from `conf.py`. Doing so interferes "
            "with Ablog's ability to stay compatible with Sphinx themes that "
            "support it out of the box. Please remove `get_html_templates_path` "
            "from `templates_path` in your `conf.py` to resolve this."
        )
```

Reply:
Oh wait, this can't be an early return since this function is also doing `blog_post_pattern` stuff.

Well, the idea still stands, although the specific code suggestion here is wrong. We'd probably want to indent the lines that add the template instead, and only inject templates if the user hasn't explicitly configured a skip.

Reply:
It might make sense to make the two of them smaller functions, which would change this function to:

```
def builder_inited(app):
    if not app.config.skip_injecting_base_ablog_templates:
        _inject_base_templates(app)
    if isinstance(app.config.blog_post_pattern, str):
        _inject_matched_blog_posts(app)
```

Reply:
I moved the `blog_post_pattern ` back to a config-init function where it used to be. This should separate the two parts nicely.